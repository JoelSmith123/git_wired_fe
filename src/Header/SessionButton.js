
import React, { Component } from 'react';

export default class Header extends Component {

  constructor(props) {
    super(props)
    this.updateState = this.props.changeLoggedInState.bind(this)
  }


 goToAuth = () => {
   window.location.href='https://git-wired-be.herokuapp.com/authenticate'
 }

 loginButton = () => {
   return (
     <input type="button"
            onClick={ () => {
              this.props.user.setGitWiredToken('123');
              this.updateState()
                // this.goToAuth()
              }
            }
            className='header-btn header-login-btn'
            value="Login"
    />
   )
 }

 logoutButton = () => {
   return (
     <input type="button"
            onClick={ () => {
              this.props.user.logout();
              this.updateState()
              }
            }
            className='header-btn header-logout-btn'
            value="Logout"
    />
   )
 }

 render() {
  return (
    <React.Fragment>
      { this.props.user.isLoggedIn() ? this.logoutButton() : this.loginButton() }
    </React.Fragment>
  )
 }

}
