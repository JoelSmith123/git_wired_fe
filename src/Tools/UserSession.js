export default class UserSession {
  constructor() {
    this.tokenKey = 'userToken'
  }

  setGitWiredToken(token) {
    sessionStorage.setItem(this.tokenKey, token);
  }

  getGitWiredToken() {
    return sessionStorage.getItem(this.tokenKey);
  }

  logout() {
    sessionStorage.removeItem(this.tokenKey);
  }


  isLoggedIn = () => {
    return Boolean( this.getGitWiredToken() )
  }




  // Make Fetch /login call
  // receive userToken

  // TO DO - TEST ME
  //  - CONFIRM endpoint
  //  - CONFIRM RESPONSE STRUCTURE
  // loginUser() {
  //   let url = 'https://git-wired-be.herokuapp.com/login'
  //   fetch(url)
  //     .then(response => response.json())
  //     .then(data     => this.getToken(data).bind(this))
  //     .then(token    => this.setGitWiredToken(token).bind(this))
  //     .catch(error   => { console.log(error); });
  // }
  //
  // getToken(data) {
  //   let token = data[this.tokenKey]
  //   return token
  // }


}
