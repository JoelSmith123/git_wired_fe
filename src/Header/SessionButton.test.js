import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import SessionButton from './SessionButton.js'
import UserSession from '../Tools/UserSession.js'

describe('SessionButton', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SessionButton user={new UserSession} changeLoggedInState={jest.fn()}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<SessionButton user={new UserSession} changeLoggedInState={jest.fn()}/>)
    expect(wrapper).toMatchSnapshot()
  }) 

  it('should return loginButton', () => {
    const wrapper = shallow(<SessionButton user={new UserSession} changeLoggedInState={jest.fn()}/>)
    const mockButton = "{\"type\":\"input\",\"key\":null,\"ref\":null,\"props\":{\"type\":\"button\",\"className\":\"header-btn header-login-btn\",\"value\":\"Login\"},\"_owner\":null,\"_store\":{}}"

    expect(JSON.stringify(wrapper.instance().loginButton())).toEqual(mockButton)
  })
})