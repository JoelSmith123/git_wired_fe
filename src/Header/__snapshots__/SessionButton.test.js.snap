// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionButton should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    changeLoggedInState={[MockFunction]}
    user={
      UserSession {
        "getURLToken": [Function],
        "isLoggedIn": [Function],
        "login": [Function],
        "tokenKey": "userToken",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <input
        className="header-btn header-login-btn"
        onClick={[Function]}
        type="button"
        value="Login"
      />,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "header-btn header-login-btn",
        "onClick": [Function],
        "type": "button",
        "value": "Login",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <input
          className="header-btn header-login-btn"
          onClick={[Function]}
          type="button"
          value="Login"
        />,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "header-btn header-login-btn",
          "onClick": [Function],
          "type": "button",
          "value": "Login",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
